src/components/socialMedia/SocialMediaContainer.js

```diff
@@ -1,26 +1,89 @@
-import React, { useState, useEffect} from "react";
+import React, { useState, useEffect, useReducer} from "react";
 import ProfileCreator from "./ProfileCreator";
 import AdBreakdown from "./AdBreakdown";
 import AdDisplay from "./AdDisplay";
 import "./socialMedia.css";
 import "../mainContent/mainContent.css";
 import "../../assets/labContent";
 import initialAds from "./initialAds.js";
+
+//Reducer setup outside of component
+const initialFreqs = [...initialAds.adList];
+const reducer = (adFrequencies, action) => {
+    switch(action.type){
+        //reset the ads back to zero
+        case "reset": {
+            return adFrequencies.map(
+                (job) => {
+                    return {...job, y : 0 }; 
+                }
+            )
+        }
+
+        case "increment": {
+            //increment just one part of the ad (more efficent than using state)
+            return adFrequencies.map(
+                (job) => {
+                    if (job.x === action.payload)
+                        return {...job,  y: job.y +1};
+                    else return job; //Should this use spread operator?
```

Comment:
I don't think we need to use the spread operator, keeping it `job` would be simpler/less redundant?